modules/pubwiseAnalyticsAdapter.js

```diff
@@ -51,72 +68,259 @@ function enrichWithUTM(dataBag) {
   try {
     for (let prop in utmKeys) {
       utmKeys[prop] = utils.getParameterByName(prop);
-      if (utmKeys[prop] != '') {
+      if (utmKeys[prop]) {
         newUtm = true;
         dataBag[prop] = utmKeys[prop];
       }
     }
 
     if (newUtm === false) {
       for (let prop in utmKeys) {
-        let itemValue = localStorage.getItem(`pw-${prop}`);
-        if (itemValue.length !== 0) {
+        let itemValue = localStorage.getItem(setNamespace(prop));
+        if (itemValue !== null && typeof itemValue !== 'undefined' && itemValue.length !== 0) {
           dataBag[prop] = itemValue;
         }
       }
     } else {
       for (let prop in utmKeys) {
-        localStorage.setItem(`pw-${prop}`, utmKeys[prop]);
+        localStorage.setItem(setNamespace(prop), utmKeys[prop]);
```

Comment:
for local storage read write, use our util methods:
-`utils.getDataFromLocalStorage(key)`
-`utils.setDataInLocalStorage(key, value)`

required by config deviceAccess flag for enforcement